cmd/mailbox-controller/controller.go

```diff
@@ -143,9 +151,18 @@ func (ctl *mbCtl) enqueue(obj any) {
 		logger.V(4).Info("Enqueuing reference due to workspace", "wsName", typed.Name)
 		ctl.queue.Add(typed.Name)
 	case *edgev2alpha1.SyncTarget:
-		mbwsName := mbwsNameOfSynctarget(typed)
-		logger.V(4).Info("Enqueuing reference due to SyncTarget", "wsName", mbwsName, "syncTargetName", typed.Name)
-		ctl.queue.Add(mbwsName)
+		mbwsName, err := ctl.mbwsNameOfSynctarget(typed)
+		if err != nil {
+			if err.Error() == errNoSpaceId.Error() {
+				logger.V(4).Info("Enqueuing SyncTarget reference for later retry", "syncTargetName", typed.Name)
+				ctl.queue.Add(refSyncTarget(typed.Name))
+			} else {
+				logger.Error(nil, "Failed to construct mailbox workspace name from SyncTarget", "syncTargetName", typed.Name)
+			}
```

Comment:
Now that all this code is duplicated elsewhere, and does not have to appear here, wouldn't it be simpler to always enqueue a `refSyncTarget`?